CoffeeChat/Controller/MatchViewController.swift

```diff
@@ -192,37 +246,82 @@ class MatchViewController: UIViewController {
 
         matchSummaryTableView.snp.makeConstraints { make in
             make.top.equalTo(matchProfileBackgroundView.snp.bottom).offset(padding)
-            if !hasReachedOut {
+            if shouldShowReachOutButton {
                 make.bottom.equalTo(reachOutButton.snp.top).offset(-padding)
             } else {
                 make.bottom.equalToSuperview().inset(padding)
             }
             make.leading.trailing.equalToSuperview().inset(padding)
         }
 
-        if !hasReachedOut {
+        if shouldShowReachOutButton {
             reachOutButton.snp.makeConstraints { make in
                 make.bottom.equalTo(view.safeAreaLayoutGuide).inset(reachOutPadding)
                 make.centerX.equalToSuperview()
                 make.size.equalTo(reachOutButtonSize)
             }
         }
+    }
+
+    private func setupMatchSummaries() {
+        guard let pair = pair else { return }
+        var matchSummary: MatchSummary
+
+        let commonInterests = user.interests.filter { pair.interests.contains($0) }
+        if commonInterests.count > 0 {
+            matchSummary = MatchSummary(title: "You both love...", detail: stringListToSentence(words: commonInterests))
+            matchSummaries.append(matchSummary)
+        }
+
+        let commonGroups = user.groups.filter { pair.groups.contains($0) }
+        if commonGroups.count > 0 {
+            matchSummary = MatchSummary(title: "You're both part of...", detail: stringListToSentence(words: commonGroups))
+            matchSummaries.append(matchSummary)
+        }
+
+        let exclusivePairInterests = pair.interests.filter { !commonInterests.contains($0) }
+        if exclusivePairInterests.count > 0 {
+            matchSummary = MatchSummary(title: "He also enjoys...", detail: stringListToSentence(words: exclusivePairInterests))
+            matchSummaries.append(matchSummary)
+        }
 
+        let exclusivePairGroups = pair.groups.filter { !commonGroups.contains($0) }
```

Comment:
Wondering if there is any way to combine these last four into a single method? But if its too much trouble, probably not a big deal.